### PR TITLE
96 remove staking contract from active contracts if rewards equals zero

### PR DESCRIPTION
**Pull request by aliberkhsl**
#96

### CODE REVIEW

File: contracts/staking/GoodFundManager.sol [hunk 1]
```diff
@@ -193,7 +193,14 @@ contract GoodFundManager is DAOContract {
 		}
 
 		if (_isBlackListed == false) {
-			if (exist == false) activeContracts.push(_stakingAddress);
+			if (exist == false && _rewardsPerBlock != 0) {
+				activeContracts.push(_stakingAddress);
+			} else {
```
Comment (sirpy) [inline contracts/staking/GoodFundManager.sol:198]: else if(exists && _rewardsPerBlock==0)
again DRY.
the whole block from if(_isBlacklisted ==false) should be changed to
```
if(exists && (_isBlacklisted || rewards==0) remove
else if(!exists && !(_isBlacklisted || rewards==0) add

Comment (sirpy) [reply contracts/staking/GoodFundManager.sol:198]: also we are waisting gas at lines 178-185 they should be moved to the else if(!exists...)

File: test/staking/StakingRewards.test.ts [hunk 2]
```diff
@@ -1234,4 +1234,45 @@ describe("StakingRewards - staking with cDAI mocks and get Rewards in GoodDollar
...
+      cBat.address,
+      BLOCK_INTERVAL,
+      nameService.address,
+      "Good BaT",
+      "gBAT",
+      "50",
+      batUsdOracle.address,
+      "100000"
+    );
+    const ictrl = await ethers.getContractAt(
```
Comment (sirpy) [inline test/staking/StakingRewards.test.ts:1254]: there's a utility genericCall please use it

File: test/staking/StakingRewards.test.ts [hunk 3]
```diff
@@ -1234,4 +1234,45 @@ describe("StakingRewards - staking with cDAI mocks and get Rewards in GoodDollar
...
+      "setStakingReward",
+      ["0", simpleStaking.address, 10, 1000, false] // set 10 gd per block
+    );
+    await ictrl.genericCall(goodFundManager.address, encodedData, avatar, 0);
+
+    const activeContractsCountAfterRemoved = await goodFundManager.getActiveContractsCount()
+
+    expect(activeContractsCountAfterAdded).to.be.gt(activeContractsCount)
+    expect(activeContractsCountAfterAdded).to.be.gt(activeContractsCountAfterRemoved)
+    expect(activeContractsCount).to.be.equal(activeContractsCountAfterRemoved)
```
Comment (sirpy) [inline test/staking/StakingRewards.test.ts:1275]: better to also test that activeContracts[latestIndex] = added contract
after remove
and activeContracts[latestIndex] = 0

**Review by sirpy — CHANGES_REQUESTED** [review]

File: contracts/staking/GoodFundManager.sol (hunk 1 above)
Comment (aliberkhsl) [reply contracts/staking/GoodFundManager.sol:198]: Fixed them

Comment (aliberkhsl) [reply contracts/staking/GoodFundManager.sol:198]: If we would move it if(!exists...) then we can't update exist ones with new reward values , blockStart or blockEnd

**Pull request merged**